src/values/dynamic-item-value.js

```diff
@@ -0,0 +1,39 @@
+import ChildElements from '../elements/child-elements';
+import FormValues from './form-values';
+
+export default class DynamicItemValue {
+  constructor(dynamicValue, attributes) {
+    this.dynamicValue = dynamicValue;
+    this._metadataJSON = attributes.metadata;
+    this._elementsJSON = attributes.elements;
+    this._valuesJSON = attributes.values;
+  }
+
+  toJSON() {
+    const json = {};
+
+    json.metadata = this._metadataJSON || null;
+    json.elements = this._elementsJSON || null;
+    json.values = this.values.toJSON() || null;
+
+    return json;
+  }
+
+  get id() {
+    return this.metadata.id;
+  }
+
+  get metadata() {
```

Comment:
we may need to add a setter for metadata (and maybe id as well)

Reply:
I think we should just modify the existing JSON object, which we can do with the getter instead of replacing the whole thing. I am adding a unit test to show that we can modify the metadata.

And I don't think we should explicitly allow updating the ID.

Reply:
Sounds good to me